narwhal/primary/src/synchronizer.rs

```diff
@@ -393,90 +411,133 @@ impl Synchronizer {
+                    let _ = result_sender.send(
+                        Self::process_certificate_with_lock(&inner, certificate, early_suspend)
+                            .await,
+                    );
+                }
+            },
+            "SynchronizerCertificateAcceptor"
+        );
 
         // Start tasks to broadcast created certificates.
         let inner_senders = inner.clone();
-        spawn_monitored_task!(async move {
-            let Ok(network) = rx_synchronizer_network.await else {
+        spawn_logged_monitored_task!(
+            async move {
+                let Ok(network) = rx_synchronizer_network.await else {
                 error!("Failed to receive Network!");
                 return;
             };
-            let mut senders = inner_senders.certificate_senders.lock();
-            for (name, _, network_key) in inner_senders
-                .committee
-                .others_primaries_by_id(inner_senders.authority_id)
-                .into_iter()
-            {
-                senders.spawn(Self::push_certificates(
-                    network.clone(),
-                    name,
-                    network_key,
-                    tx_own_certificate_broadcast.subscribe(),
-                ));
-            }
-            if let Some(cert) = highest_created_certificate {
-                // Error can be ignored.
-                if tx_own_certificate_broadcast.send(cert).is_err() {
-                    error!("Failed to populate initial certificate to send to peers!");
+                let mut senders = inner_senders.certificate_senders.lock();
+                for (name, _, network_key) in inner_senders
+                    .committee
+                    .others_primaries_by_id(inner_senders.authority_id)
+                    .into_iter()
+                {
+                    senders.spawn(Self::push_certificates(
+                        network.clone(),
+                        name,
+                        network_key,
+                        tx_own_certificate_broadcast.subscribe(),
+                    ));
                 }
-            }
-        });
+                if let Some(cert) = highest_created_certificate {
+                    // Error can be ignored.
+                    if tx_own_certificate_broadcast.send(cert).is_err() {
+                        error!("Failed to populate initial certificate to send to peers!");
+                    }
+                }
+            },
+            "SynchronizerCertificateBroadcasters"
+        );
+
+        // Start a task to async download batches if needed
+        let weak_inner = Arc::downgrade(&inner);
+        spawn_logged_monitored_task!(
+            async move {
+                let mut batch_tasks: JoinSet<DagResult<()>> = JoinSet::new();
+
+                loop {
+                    tokio::select! {
+                        Some((header, max_age, certified)) = rx_batch_tasks.recv() => {
```

Comment:
This task should also shutdown when rx_batch_tasks.recv() is None.

Reply:
you are right!